backend/src/apis/ownership.rs

```diff
@@ -0,0 +1,37 @@
+use crate::contracts::{generated::summa_contract::AddressOwnershipProof, signer::SummaSigner};
+use ethers::types::Address;
+use std::{error::Error, result::Result};
+
+pub struct AddressOwnership {
+    address_ownership_proofs: Vec<AddressOwnershipProof>,
+    signer: SummaSigner,
+}
+
+impl AddressOwnership {
+    pub fn new(
+        main_signer_key: &str,
+        chain_id: u64,
+        rpc_url: &str,
+        address: Address,
+    ) -> Result<AddressOwnership, Box<dyn Error>> {
+        Ok(AddressOwnership {
+            address_ownership_proofs: vec![],
+            signer: SummaSigner::new(&[], main_signer_key, chain_id, rpc_url, address),
+        })
+    }
+
+    // Make sure the input `addresses` in `address_ownership_proof` are not duplicated with addresses already registered on the contract.
```

Comment:
For what I can tell, this operation of making sure that the input 'addresses` ... is not being performed now so the comment is misleading. BTW I agree that we shouldn't do that.

Reply:
Good point, that's why I also think that the address ownership should live inside the Round and Round should take care of it. (cc @sifnoc )

Reply:
I'm not sure if they'll be performed separately in the real-world implementation. E.g., CEX can add more addresses in between the proving rounds but in each round it has to check if all the addresses have the ownership proofs submitted, so it makes sense just to submit new addresses in the beginning of the round. BTW I agree that it's good to separate address ownership into its own trait.

Reply:
Updated comment at [ce908f0](https://github.com/summa-dev/summa-solvency/pull/152/commits/ce908f0a7323153aea8f3f959f16e69766a661d7)

Both perspectives are valid. However, from my understanding, the `AddressOwnership` operation isn't intrinsically tied to `Round`. As such, it might not be fitting to merge `AddressOwnership` directly into `Round` at this juncture.
That being said, there could be value in creating a new struct that encompasses both functionalities. 
Perhaps something like `SummaOperator` or `SummaController`?
For a CEX operator, a consolidated interface for both `Round` and `AddressOwnership` would undoubtedly be beneficial.

Reply:
I'm not talking about merging (similarly to how Snapshot can still live in a separate file and not inside the same file with Round). I'm only talking about the encapsulation of AddressOwnership inside Round (Round holding an instance of AddressOwnership). As I imagine the future use of the system, CEX will not call any of Summa services from within CEX services, they will be called by the scheduler. If everything gets called with the same interval, it's all a "Round" of a proof of solvency.

Reply:
I now understand your perspective. You're suggesting a `Round` structure similar to this:
```Rust
pub struct Round<const LEVELS: usize, const N_ASSETS: usize, const N_BYTES: usize> {
    timestamp: u64,
    snapshot: Snapshot<LEVELS, N_ASSETS, N_BYTES>,
    address_ownership: AddressOwnership,
    signer: SummaSigner,
}
```

Upon reviewing the following code segment, I see its alignment with your viewpoint on AddressOwnership:
https://github.com/summa-dev/summa-solvency/blob/c4edbc6fe75131690c9f323be333e2d5c12e20df/contracts/src/Summa.sol#L107-L110

However, regardless of the naming convention for `Round`, I believe there should be only one `Signer`. 
If there are multiple signers, they might generate transactions with the same nonce.

Reply:
> I'm only talking about the encapsulation of AddressOwnership inside Round (Round holding an instance of AddressOwnership). As I imagine the future use of the system, CEX will not call any of Summa services from within CEX services, they will be called by the scheduler. If everything gets called with the same interval, it's all a "Round" of a proof of solvency.

My hypothesis is that `Round` and `AddressOwnership` are two very separate things and the Scheduler will only have to deal with `Round`. The soundness of this hypothesis depends on the frequency of address rotation performed by an Exchange. If they stick to the same address for long time, `AddressOwnership` is just called once and then just updated once in a while. On the contrary, if the switch addresses very frequently, it's likely that at every round a new proof of address ownership must be called.

I think we can stick to this now and challenge this hypothesis as soon as speak with customers. This will also inform the design of the Summa services.

Reply:
> However, regardless of the naming convention for `Round`, I believe there should be only one `Signer`.
> If there are multiple signers, they might generate transactions with the same nonce.

Can elaborate on that @sifnoc ?

Reply:
In the current implementation, both `Round` and `AddressOwnership` have an instance of Signer, each. 
The default middleware in `ethers-rs` does not provide a shared nonce counter between signers. 
As a result, both signers might query the node using `get_transaction_count` for the same `Account`. 

This can lead to a high likelihood that both signers will generate transactions with the same nonce for `submit_proof_of_solvency` and `submit_proof_of_address_ownership`. 

Consequently, one of these transactions may not be included in the block.

Reply:
Yes, the same Signer should be injected everywhere where needed.